src/spec.rs

```diff
@@ -679,6 +696,11 @@ impl Heatmap {
             };
             self.domain = Some(domain);
         }
+        if let Some(domain_mid) = &self.domain_mid {
+            let old_domain = self.domain.as_mut().unwrap();
+            old_domain.insert(1, domain_mid.to_string());
```

Comment:
THis now does not really pass mid to vega, but instead modifies the domain? Is that what David meant?

Reply:
Ah wait, I guess this is because you anyway auto-infer the min and the max?

Reply:
But how would this play with the user also specifying two values for range?

Reply:
> Ah wait, I guess this is because you anyway auto-infer the min and the max?

Yes this in combination with the fact that the scale api from vega isn't documented well enough to figure out how one would even pass `domainMid` to the scale function. I've tried multiple ways that would make sense to me but neither one worked.

Davids problem was indeed that he didn't want to specify a domain himself so i think this should solve his problem i think.

> But how would this play with the user also specifying two values for range?

Well datavzrd doesn't crash but the result doesn't really make sense. Should i add a check to the range to have a length of 3 when `mid-domain` is specified?

Reply:
> Well datavzrd doesn't crash but the result doesn't really make sense. Should i add a check to the range to have a length of 3 when mid-domain is specified?
Yes, that sounds like the perfect fix.

Reply:
Alrighty i am on it!

Reply:
Solved in https://github.com/datavzrd/datavzrd/pull/574/commits/0c45156c8b72c99d29fb7405c902ce6ead394a3c